### PR TITLE
fix: `open_url` not working on windows

### LINKED ISSUE #3128: Submit a bug report on Github debug button fails to generate report
### Describe the bug

When pressing the debug option to open github to file a report, it correctly opens the window but then reports that it fails, and doesn't fill out the report details, it adds OS version game version and then fails.

I have no idea if this happens without mods, but I can investigate that too if needed.

### Steps To Reproduce

1. Use my my mod list and game version.
2. Activate the function.
3. It fails.

### Screenshots

_No response_

### Versions and configuration

- OS: Windows
    - OS Version: Win 10 Pro for Workstations 64 bit build 19045
- Game Version: e01f656 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en_US]
- Mods loaded: [
    Bright Nights [bn],
    Elevated bridges [elevated_bridges],
    Craftable Gun Pack [craftgp],
    Alternative Map Key [alt_map_key],
    Fuji's Military Profession Pack [fuji_mpp],
    Arcana and Magic Items [Arcana],
    Craftable Bullet Casings [CasingCrafting],
    MST Extra [MST_Extra],
    Tankmod: Revived [Tankmod_Revived],
    Essence 2200 [Essence2200],
    Essence 2200 Tweaks [FoxTweaks],
    Nonperishable Overhaul [Nonperishable_Overhaul],
    No Addictions [No_Addictions],
    Limit Fungal Growth [limit_fungal_growth],
    Disable NPC Needs [no_npc_food],
    Urban Development [Urban_Development],
    Project: Pack Rat [Jake's_Squirrel_mod]
]

### Additional context

 DEBUG    : Failed to open URL: https://github.com/cataclysmbnteam/Cataclysm-BN/issues/new?labels=bug&template=bug_report.yml&versions-and-configuration=-%20OS%3A%20Windows%0A%20%20%20%20-%20OS%20Version%3A%2010.0%202009%0A-%20Game%20Version%3A%20e01f656%20%5B64-bit%5D%0A-%20Graphics%20Version%3A%20Tiles%0A-%20Game%20Language%3A%20English%20%5Ben_US%5D%0A-%20Mods%20loaded%3A%20%5B%0A%20%20%20%20Bright%20Nights%20%5Bbn%5D%2C%0A%20%20%20%20Elevated%20bridges%20%5Belevated_bridges%5D%2C%0A%20%20%20%20Craftable%20Gun%20Pack%20%5Bcraftgp%5D%2C%0A%20%20%20%20Alternative%20Map%20Key%20%5Balt_map_key%5D%2C%0A%20%20%20%20Fuji%27s%20Military%20Profession%20Pack%20%5Bfuji_mpp%5D%2C%0A%20%20%20%20Arcana%20and%20Magic%20Items%20%5BArcana%5D%2C%0A%20%20%20%20Craftable%20Bullet%20Casings%20%5BCasingCrafting%5D%2C%0A%20%20%20%20MST%20Extra%20%5BMST_Extra%5D%2C%0A%20%20%20%20Tankmod%3A%20Revived%20%5BTankmod_Revived%5D%2C%0A%20%20%20%20Essence%202200%20%5BEssence2200%5D%2C%0A%20%20%20%20Essence%202200%20Tweaks%20%5BFoxTweaks%5D%2C%0A%20%20%20%20Nonperishable%20Overhaul%20%5BNonperishable_Overhaul%5D%2C%0A%20%20%20%20No%20Addictions%20%5BNo_Addictions%5D%2C%0A%20%20%20%20Limit%20Fungal%20Growth%20%5Blimit_fungal_growth%5D%2C%0A%20%20%20%20Disable%20NPC%20Needs%20%5Bno_npc_food%5D%2C%0A%20%20%20%20Urban%20Development%20%5BUrban_Development%5D%2C%0A%20%20%20%20Project%3A%20Pack%20Rat%20%5BJake%27s_Squirrel_mod%5D%0A%5D%0A
Attemped command was: explorer "https://github.com/cataclysmbnteam/Cataclysm-BN/issues/new?labels=bug&template=bug_report.yml&versions-and-configuration=-%20OS%3A%20Windows%0A%20%20%20%20-%20OS%20Version%3A%2010.0%202009%0A-%20Game%20Version%3A%20e01f656%20%5B64-bit%5D%0A-%20Graphics%20Version%3A%20Tiles%0A-%20Game%20Language%3A%20English%20%5Ben_US%5D%0A-%20Mods%20loaded%3A%20%5B%0A%20%20%20%20Bright%20Nights%20%5Bbn%5D%2C%0A%20%20%20%20Elevated%20bridges%20%5Belevated_bridges%5D%2C%0A%20%20%20%20Craftable%20Gun%20Pack%20%5Bcraftgp%5D%2C%0A%20%20%20%20Alternative%20Map%20Key%20%5Balt_map_key%5D%2C%0A%20%20%20%20Fuji%27s%20Military%20Profession%20Pack%20%5Bfuji_mpp%5D%2C%0A%20%20%20%20Arcana%20and%20Magic%20Items%20%5BArcana%5D%2C%0A%20%20%20%20Craftable%20Bullet%20Casings%20%5BCasingCrafting%5D%2C%0A%20%20%20%20MST%20Extra%20%5BMST_Extra%5D%2C%0A%20%20%20%20Tankmod%3A%20Revived%20%5BTankmod_Revived%5D%2C%0A%20%20%20%20Essence%202200%20%5BEssence2200%5D%2C%0A%20%20%20%20Essence%202200%20Tweaks%20%5BFoxTweaks%5D%2C%0A%20%20%20%20Nonperishable%20Overhaul%20%5BNonperishable_Overhaul%5D%2C%0A%20%20%20%20No%20Addictions%20%5BNo_Addictions%5D%2C%0A%20%20%20%20Limit%20Fungal%20Growth%20%5Blimit_fungal_growth%5D%2C%0A%20%20%20%20Disable%20NPC%20Needs%20%5Bno_npc_food%5D%2C%0A%20%20%20%20Urban%20Development%20%5BUrban_Development%5D%2C%0A%20%20%20%20Project%3A%20Pack%20Rat%20%5BJake%27s_Squirrel_mod%5D%0A%5D%0A"

 FUNCTION : open_url
 FILE     : D:\a\Cataclysm-BN\Cataclysm-BN\src\url_utility.cpp
 LINE     : 26
 VERSION  : BN e01f656

### PR DESCRIPTION
**Pull request by scarf005**
#### Summary

SUMMARY: Bugfixes "Made 'Submit a bug report on Github debug' work correctly on windows"

#### Purpose of change

- fixes #3128

#### Describe the solution

https://chat.openai.com/share/af7046b4-832d-467f-a1d9-700546554c31

#### Testing

manually typing the command on CMD.exe worked

https://github.com/cataclysmbnteam/Cataclysm-BN/issues/3128#issuecomment-1712982005

### CODE REVIEW

**Comment by olanti-p** [comment]
> https://chat.openai.com/share/af7046b4-832d-467f-a1d9-700546554c31

That link is dead or private.

Also, LLMs are not always correct when it comes to code, their results should be double-checked.

File: src/url_utility.cpp [hunk 1]
```diff
@@ -8,10 +8,8 @@
 void open_url( const std::string &url )
 {
     static const std::string executable =
-#if defined(_WIN32)
-        "explorer";
-#elif defined(_WIN64)
-        "start";
+#if defined(_WIN32) || defined(_WIN64)
+        "start \"\"";
```
Comment (olanti-p) [inline src/url_utility.cpp:12]: `_WIN32` is defined for both x32 and x64 targets, so check for `_WIN64` is redundant here
https://learn.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=msvc-170

**Comment by scarf005** [comment]
image
strange, i can access it from private mode.

https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/start suggests `start` can be used on both 32bit and 64bit machine.

**Review by olanti-p — APPROVED** [review]

**Comment by scarf005** [comment]
doesn't seem to work, #3321

**Pull request merged**